plugin/kubernetes/controller.go

```diff
@@ -173,6 +175,30 @@ func newdnsController(ctx context.Context, kubeClient kubernetes.Interface, opts
 	return &dns
 }
 
+func cleanEndpointSlice(i interface{}) {
+	if e, ok := i.(*discovery.EndpointSlice); ok {
+		*e = discovery.EndpointSlice{}
+	}
+}
```

Comment:
I think this would be functionally the same, but is it worth making a `clean` function that does a type switch? That'd keep the informer.DefaultProcessor function parameters unchanged.

```
func cleanup(I interface{}) {
  switch item := i.(type) {
  case *api.Endpoints:
    *item = api.Endpoints{}
  case *api.Service:
    *item = api.Service{}
  ...
  }
}
```

Reply:
Yes, thats better.

Reply:
done, thanks!